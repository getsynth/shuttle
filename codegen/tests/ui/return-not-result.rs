#[shuttle_codegen::main]
async fn return_not_result() -> bool {}
