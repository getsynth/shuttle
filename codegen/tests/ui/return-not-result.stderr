error: shuttle_service::main functions need to return a first class service or 'Result<impl Service, shuttle_service::Error>

         = help: See the docs for services with first class support
         = note: https://docs.rs/shuttle-service/latest/shuttle_service/attr.main.html#shuttle-supported-services

 --> tests/ui/return-not-result.rs:2:33
  |
2 | async fn return_not_result() -> bool {}
  |                                 ^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui/return-not-result.rs:2:40
  |
2 | async fn return_not_result() -> bool {}
  |                                        ^ consider adding a `main` function to `$DIR/tests/ui/return-not-result.rs`
